lib/logstash/inputs/imap.rb

```diff
@@ -149,6 +150,18 @@ def check_mail(queue)
     end
   end
 
+  def parse_attachments(mail)
+    attachments = []
+    mail.attachments.each do |attachment|
+      if @save_attachments
+        attachments << { "filename" => attachment.filename, "data" => attachment.body.encoded }
```

Comment:
This may not produce `Base64` encoded data depending on the `Content-Transfer-Encoding` in the message (eg, run the unit test with a text attachement - ```add_file :filename => "some.txt", :content => msg_text```)

Reply:
Good point, have added a test for 7BIT encoding based on https://www.w3.org/Protocols/rfc1341/5_Content-Transfer-Encoding.html.
I tested with 7BIT, 8BIT and binary transfer encoding options with data read from /bin/cp and didn't see any difference.